bitwarden_license/src/Commercial.Core/SecretsManager/Commands/AccessPolicies/DeleteAccessPolicyCommand.cs

```diff
@@ -1,27 +1,139 @@
-﻿using Bit.Core.Exceptions;
+﻿using Bit.Core.Context;
+using Bit.Core.Enums;
+using Bit.Core.Exceptions;
 using Bit.Core.SecretsManager.Commands.AccessPolicies.Interfaces;
+using Bit.Core.SecretsManager.Entities;
 using Bit.Core.SecretsManager.Repositories;
 
 namespace Bit.Commercial.Core.SecretsManager.Commands.AccessPolicies;
 
 public class DeleteAccessPolicyCommand : IDeleteAccessPolicyCommand
 {
     private readonly IAccessPolicyRepository _accessPolicyRepository;
+    private readonly ICurrentContext _currentContext;
+    private readonly IProjectRepository _projectRepository;
+    private readonly IServiceAccountRepository _serviceAccountRepository;
 
-    public DeleteAccessPolicyCommand(IAccessPolicyRepository accessPolicyRepository)
+    public DeleteAccessPolicyCommand(
+        IAccessPolicyRepository accessPolicyRepository,
+        ICurrentContext currentContext,
+        IProjectRepository projectRepository,
+        IServiceAccountRepository serviceAccountRepository)
     {
+        _projectRepository = projectRepository;
+        _serviceAccountRepository = serviceAccountRepository;
         _accessPolicyRepository = accessPolicyRepository;
+        _currentContext = currentContext;
     }
 
 
-    public async Task DeleteAsync(Guid id)
+    public async Task DeleteAsync(Guid id, Guid userId)
     {
         var accessPolicy = await _accessPolicyRepository.GetByIdAsync(id);
         if (accessPolicy == null)
         {
             throw new NotFoundException();
         }
 
-        await _accessPolicyRepository.DeleteAsync(id);
+        switch (accessPolicy)
+        {
+            case UserProjectAccessPolicy ap:
+                if (ap.GrantedProject == null)
+                {
+                    throw new NotFoundException();
+                }
```

Comment:
The check is there because it's a nullable field, and we get warnings if we don't check for null.

It's nullable because when we are mapping requests onto the model we won't populate the field. 

When mapping database queries, the field should never be null.

Not sure what you mean by a validate method.

Reply:
Oh right. You can ignore the warnings using `!`, or null coalece with `?`.

Reply:
Resolved in https://github.com/bitwarden/server/pull/2628/commits/79bce819b50d75b5ce021b4b4dfad21dc9c97cd8